build/scoring_test.go

```diff
@@ -60,7 +60,59 @@ func TestFileNameMatch(t *testing.T) {
 	}
 
 	for _, c := range cases {
-		checkScoring(t, c, ctags.UniversalCTags)
+		checkScoring(t, c, false, ctags.UniversalCTags)
+	}
+}
+
+func TestBM25(t *testing.T) {
+	exampleJava, err := os.ReadFile("./testdata/example.java")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cases := []scoreCase{
+		{
+			// Matches on both filename and content
+			fileName: "example.java",
+			query:    &query.Substring{Pattern: "example"},
+			content:  exampleJava,
+			language: "Java",
+			// keyword-score:1.63 (sum-tf: 6.00, length-ratio: 2.00)
+			wantScore: 1.63,
```

Comment:
Not a blocker, but I think it would be nice if the code added debug information or something similar. It's hard for me to tell whether this actually matches filename and content. 

The existing tests in this file have the same issue, but the scores are easier to interpret.

Reply:
Ah, I see `sum-tf` should do just that.

Reply:
If I understand right, we're all good here. I also feel `sum-tf` gives enough information, and it's not worth tracking a further breakdown into filename tf and content tf.